src/types/mod.rs

```diff
@@ -28,4 +29,7 @@ mod utils;
 ///
 /// While `T` does not have to implement `Composite` to be used with `Shared`,
 /// the primary use of `Shared` is with types that implement `Composite`.
+#[cfg(feature = "client")]
 pub type Shared<T> = Arc<RwLock<T>>;
+#[cfg(not(feature = "client"))]
+pub type Shared<T> = T;
```

Comment:
Seems a bit hacky, I do see why you did it like this however
maybe document this though, something like
```suggestion
///
/// While `T` does not have to implement `Composite` to be used with `Shared`,
/// the primary use of `Shared` is with types that implement `Composite`.
///
/// When the `client` feature is disabled, this does nothing (same as just `T`),
/// since `Composite` structures are disabled.
#[cfg(feature = "client")]
pub type Shared<T> = Arc<RwLock<T>>;
#[cfg(not(feature = "client"))]
pub type Shared<T> = T;
```